client/app/pages/groups/GroupDataSources.jsx

```diff
@@ -0,0 +1,262 @@
+import { isBoolean, filter, map } from 'lodash';
+import React from 'react';
+import { react2angular } from 'react2angular';
+import Button from 'antd/lib/button';
+import Dropdown from 'antd/lib/dropdown';
+import Menu from 'antd/lib/menu';
+import Icon from 'antd/lib/icon';
+
+import { Paginator } from '@/components/Paginator';
+
+import { wrap as liveItemsList, ControllerType } from '@/components/items-list/ItemsList';
+import { ResourceItemsSource } from '@/components/items-list/classes/ItemsSource';
+import { StateStorage } from '@/components/items-list/classes/StateStorage';
+
+import LoadingState from '@/components/items-list/components/LoadingState';
+import ItemsTable, { Columns } from '@/components/items-list/components/ItemsTable';
+import SelectItemsDialog from '@/components/SelectItemsDialog';
+import { DataSourcePreviewCard } from '@/components/PreviewCard';
+
+import GroupName from '@/components/groups/GroupName';
+import ListItemAddon from '@/components/groups/ListItemAddon';
+import Sidebar from '@/components/groups/DetailsPageSidebar';
+
+import { $http } from '@/services/ng';
+import { currentUser } from '@/services/auth';
+import { Group } from '@/services/group';
+import { DataSource } from '@/services/data-source';
+import navigateTo from '@/services/navigateTo';
+import { routesToAngularRoutes } from '@/lib/utils';
+
+class GroupDataSources extends React.Component {
+  static propTypes = {
+    controller: ControllerType.isRequired,
+  };
+
+  groupId = parseInt(this.props.controller.params.groupId, 10);
```

Comment:
This needs to be an integer for `Group.get`?

Reply:
Also, I didn't know it was safe to read props like this! 😮😅

Reply:
I needed it to be integer in another controller (https://github.com/getredash/redash/pull/3411/files#diff-466f2851617caa71f7d6f5088cc8eaefR101), and here it's also converted because of copy-paste. But I think it's not an issue.

> Also, I didn't know it was safe to read props like this!

No sure why, but it works even if there is a constructor :thinking: